src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -829,7 +829,20 @@ internal override ManagedKind GetManagedKind(ref CompoundUseSiteInfo<AssemblySym
 
         internal bool IsFileLocal => HasFlag(DeclarationModifiers.File);
 
-        internal sealed override SyntaxTree? AssociatedSyntaxTree => IsFileLocal ? declaration.Declarations[0].Location.SourceTree : null;
+        internal SyntaxTree AssociatedSyntaxTree => declaration.Declarations[0].Location.SourceTree;
+
+        internal sealed override FileIdentifier? AssociatedFileIdentifier
+        {
+            get
+            {
+                if (!IsFileLocal)
+                {
+                    return null;
+                }
+
+                return FileIdentifier.Create(AssociatedSyntaxTree);
```

Comment:
>FileIdentifier.Create(AssociatedSyntaxTree); [](http://example.com/codeflow?start=23&length=44)

Should this value be cached?

Reply:
I think it might be good to cache it on the CSharpSyntaxTree.

Reply:
Opened #63706.